src/main/java/com/sap/sgs/phosphor/fosstars/data/github/UsesOwaspDependencyScan.java

```diff
@@ -0,0 +1,318 @@
+   */
+  private static Status availablityFrom(Set<Location> locations) {
+    if (locations == null || locations.isEmpty()) {
+      return NOTFOUND;
+    }
+
+    if (mandatoryCheckFor(locations)) {
+      return MANDATORY;
+    }
+
+    return OPTIONAL;
+  }
+
+  /**
+   * Check if plugin configurations have an attribute which can fail the build on finding
+   * vulnerabilities above a given cvss score. Hence, look for possible cvss score value which
+   * indicates, when to fail the build. The cvss score is in between 0.0 <= x <= 10.0. If the 
+   * score is > 11.0, then no vulnerabilities are considered to fail the build as this value 
+   * is out of bound.
+   *
+   * @param configuration The plugin configuration to be checked.
+   * @return The cvss score which indicates how the build should fail.
+   */
+  private static double cvssScoreFrom(Object configuration) {
+    // The OWASP dependency plugin is found and checking configurations.
+    for (String config : FAIL_BUILD_CONFIGURATIONS.keySet()) {
+      Optional<String> content = domParser(config, configuration);
+      String type = FAIL_BUILD_CONFIGURATIONS.get(config);
+
+      if (content.isPresent()) {
+        return parseScore(content.get(), type);
+      }
+    }
+
+    return OUT_OF_BOUND_CVSS_SCORE;
+  }
+
+  /**
+   * Parse the value depending on the given type. The type can be 'boolean' or 'number'. If
+   * 'boolean', then return 0.0, otherwise parse the given string value to 'number'.
+   * 
+   * @param value the value to be parsed.
+   * @param type the type of the string value.
+   * @return the parsed value.
+   */
+  private static double parseScore(String value, String type) {
+    if ("number".equals(type)) {
+      return Double.parseDouble(value);
+    }
+    return 0.0;
+  }
+
+  /**
+   * Get the value if a configuration exists.
+   * 
+   * @param configName The name of the configuration to find.
+   * @param configuration The list of configurations associated to OWASP Dependency check plugin.
+   * @return The value of the configuration if found.
+   */
+  private static Optional<String> domParser(String configName, Object configuration) {
+    if (configuration == null) {
+      return Optional.empty();
+    }
+
+    Xpp3Dom dom = ((Xpp3Dom) configuration).getChild(configName);
+    if (dom == null) {
+      return Optional.empty();
+    }
+
+    return Optional.ofNullable(dom.getValue());
+  }
+
+  /**
+   * Returns whether the plugin will be executed as a mandatory step during the build.
+   * 
+   * @param defaultParam The location of the plugin.
+   * @param locations The set of {@link Location}.
+   * @return True if the plugin will be executed as a mandatory step, false otherwise.
+   */
+  private static boolean mandatoryCheckFor(Set<Location> locations) {
+    return (locations.contains(BUILD) || locations.contains(REPORTING))  
+        && !locations.contains(PROFILE)
+        && !locations.contains(MANAGEMENT);
```

Comment:
If the plugin is mentioned in both the default build section and plugin management, then the method returns false which seems to be wrong.